tests/unit/NetworkTest.js

```diff
@@ -209,3 +221,47 @@ test('consecutive API calls eventually succeed when authToken is expired', () =>
             expect(chatList).toEqual(TEST_CHAT_LIST);
         });
 });
+
+test('retry network request if auth and credentials are not read from Onyx yet', () => {
+    const ONYX_DELAY_MS = 3000;
+    const NETWORK_INTERVAL_MS = 1000;
```

Comment:
This can be changed by `Network` so I was thinking that we'd add this in `Network.js`

Reply:
> > Hmm I don't think so, because the Network.setIsReady(true) should be only called when the app is booting up to avoid the race condition between storage and network.
> 
> Yeah I was thinking that when you sign out the storage is cleared so the values for `session` and `credentials` look like this:
> 
> <img alt="2021-11-04_12-41-48" width="507" src="https://user-images.githubusercontent.com/32969087/140430271-59e7bf7f-1d3d-4920-8ace-5f99af18076e.png">
> 
> But since we "cache-first" there's no way for an `Onyx.set()` or `Onyx.merge()` to cause a "slow" `Onyx.connect()` on sign in.
> 
> https://github.com/Expensify/react-native-onyx/blob/224e2ba03f2365726b8c1d0785e750c54a85332b/lib/Onyx.js#L530-L537

Ah yeah but even if we don't read the values from the cache after signing out, I don't think we should have a problem; the storage and the network are ready and the network loop should handle any bad state.

Reply:
I added the `PROCESS_REQUEST_DELAY_MS` const to `Network.js` and it's imported in `NetworkTest.js`